src/libraries/System.Private.CoreLib/src/System/IO/File.cs

```diff
@@ -181,45 +181,87 @@ internal static DateTimeOffset GetUtcDateTimeOffset(DateTime dateTime)
 
         public static void SetLastAccessTimeUtc(string path, DateTime lastAccessTimeUtc)
-            => FileSystem.SetLastAccessTime(Path.GetFullPath(path), GetUtcDateTimeOffset(lastAccessTimeUtc), asDirectory: false);
+            => FileSystem.SetLastAccessTime(Path.GetFullPath(path), GetUtcDateTimeOffset(lastAccessTimeUtc), false);
+
+        public static void SetLastAccessTimeUtc(SafeFileHandle fileHandle, DateTime lastAccessTimeUtc)
+            => FileSystem.SetLastAccessTime(fileHandle, GetUtcDateTimeOffset(lastAccessTimeUtc));
 
         public static DateTime GetLastAccessTime(string path)
             => FileSystem.GetLastAccessTime(Path.GetFullPath(path)).LocalDateTime;
 
+        public static DateTime GetLastAccessTime(SafeFileHandle fileHandle)
+            => FileSystem.GetLastAccessTime(fileHandle).LocalDateTime;
+
         public static DateTime GetLastAccessTimeUtc(string path)
             => FileSystem.GetLastAccessTime(Path.GetFullPath(path)).UtcDateTime;
 
+        public static DateTime GetLastAccessTimeUtc(SafeFileHandle fileHandle)
+            => FileSystem.GetLastAccessTime(fileHandle).UtcDateTime;
+
         public static void SetLastWriteTime(string path, DateTime lastWriteTime)
-            => FileSystem.SetLastWriteTime(Path.GetFullPath(path), lastWriteTime, asDirectory: false);
+            => FileSystem.SetLastWriteTime(Path.GetFullPath(path), lastWriteTime, false);
+
+        public static void SetLastWriteTime(SafeFileHandle fileHandle, DateTime lastWriteTime)
+            => FileSystem.SetLastWriteTime(fileHandle, lastWriteTime);
 
         public static void SetLastWriteTimeUtc(string path, DateTime lastWriteTimeUtc)
-            => FileSystem.SetLastWriteTime(Path.GetFullPath(path), GetUtcDateTimeOffset(lastWriteTimeUtc), asDirectory: false);
+            => FileSystem.SetLastWriteTime(Path.GetFullPath(path), GetUtcDateTimeOffset(lastWriteTimeUtc), false);
+
+        public static void SetLastWriteTimeUtc(SafeFileHandle fileHandle, DateTime lastWriteTimeUtc)
+            => FileSystem.SetLastWriteTime(fileHandle, GetUtcDateTimeOffset(lastWriteTimeUtc));
 
         public static DateTime GetLastWriteTime(string path)
             => FileSystem.GetLastWriteTime(Path.GetFullPath(path)).LocalDateTime;
 
+        public static DateTime GetLastWriteTime(SafeFileHandle fileHandle)
+            => FileSystem.GetLastWriteTime(fileHandle).LocalDateTime;
+
         public static DateTime GetLastWriteTimeUtc(string path)
             => FileSystem.GetLastWriteTime(Path.GetFullPath(path)).UtcDateTime;
 
+        public static DateTime GetLastWriteTimeUtc(SafeFileHandle fileHandle)
+            => FileSystem.GetLastWriteTime(fileHandle).UtcDateTime;
+
         public static FileAttributes GetAttributes(string path)
             => FileSystem.GetAttributes(Path.GetFullPath(path));
 
+        public static FileAttributes GetAttributes(SafeFileHandle fileHandle)
```

Comment:
if you pass a null handle this method (and probably others) throws NullReferenceException. You should check for null and throw ArgumentNullException instead. You also need to add tests for the null case.